Be more specific about the reader needs to do

jpaugh opened the pull request:
These changes reduce ambiguity, and give the reader a clearer picture of what they're supposed to do.

* Specify what kind of scene the reader needs to create
* Remove ambiguous reference to two project roots
* Reduce confusion about `res:///` paths when the user chooses a different folder for their `.gdextension`

jpaugh committed:
Give the reader detail about what kind of scene to create

- Remove confusing reference to a "default" scene, which does not appear
  to exist in Godot 4.2.
- Guide the user to create the kind of scene we'll need later on
- Guide the user to change a Sprite2D into a Player before running it
  for the first time

jpaugh committed:
Remove ambigous reference to a second project root

Since we're recommending a file structure which is a super-set of
Godot's project structure, we need to use that language consistently.

jpaugh committed:
Clear up ambiguity about file paths

This confused me, at least.

jpaugh committed:
lint: Capitalize Godot and wrap folder names in backticks

Bromeon commented on src/intro/hello-world.md:
```diff
@@ -125,8 +128,8 @@ $ ls -l
-First, add an empty `.gdextension` file anywhere in your project folder. In case you're familiar with Godot 3, this is the equivalent of
-`.gdnlib`. In this case, we create `res://HelloWorld.gdextension` in the root folder of the project and fill it as follows:
+First, add an empty `.gdextension` file anywhere in your `godot` subfolder. In case you're familiar with Godot 3, this is the equivalent of
+`.gdnlib`. In this case, we create `res://HelloWorld.gdextension` at `project_dir/godot` and fill it as follows:
```
> at `project_dir/godot` 

This is not very obvious, maybe something like "directly inside the `godot` directory"?

The term "project directory" is ambiguous (could mean the godot-rust project or the Godot project), so it might be better to avoid it here.

jpaugh replied:
Okay. I changed it to be consistent. Currently switching it to say "`godot` subfolder", since it also says that at line 47 when the folder structure is introduced. As for folder vs directory, folder is used 8 times and directory is used 5 times. Maybe making it consistent would be worthwhile. :shrug:

Bromeon commented on src/intro/hello-world.md:
```diff
@@ -44,8 +44,11 @@ project_dir
-Save the changes and consider versioning each step of the tutorial in Git.
+Open the Godot project manager and create a new Godot 4 project in the `godot/` subfolder. Add a Sprite2D to the center of a new scene.
+We recommend that you follow the [Official tutorial][tutorial-begin] and stop at the point where it asks you to create a script.
+We will show you how to do that part in Rust.
```
Add backticks around code/API symbols:
```suggestion
Open the Godot project manager and create a new Godot 4 project in the `godot/` subfolder. Add a `Sprite2D` to the center of a new scene.
We recommend that you follow the [Official tutorial][tutorial-begin] and stop at the point where it asks you to create a script.
We will show you how to do that part in Rust.
```

Bromeon commented on src/intro/hello-world.md:
````diff
@@ -279,6 +282,7 @@ Let's break this down.
 ```admonish warning title="Correct node type"
 When adding an instance of your `Player` class to the scene, make sure to select node type `Player` **and not its base `Sprite2D`**.
 Otherwise, your Rust logic will not run.
+We will guide you to make that change to your scene later, when you're ready to test it.
````
This doesn't add much, I'd remove it.

We should keep warning boxes as short and to-the-point as possible, otherwise they become inflationary. Text inside boxes typically comes in the form of isolated explanations and not the typical tutorial flow (first we do this, later this...).

Bromeon commented on src/intro/hello-world.md:
```diff
@@ -350,7 +354,12 @@ Do not use the `self.base` field directly. Use `self.base()` or `self.base_mut()
-This is a point where you can compile your code, launch Godot and see the result. The sprite should rotate at a constant speed.
+This is a point where you can see the result. Compile your code and launch the Godot editor.
+Right click on your Sprite2D in the scene tree, and choose "Change Type..."
+Find and choose the _Player_ node type, which will be a child of Sprite2D in the Change Type dialog that appears.
```
Here as well symbols in backticks (not italics) 🙂

Bromeon commented on src/intro/hello-world.md:
```diff
@@ -350,7 +354,12 @@ Do not use the `self.base` field directly. Use `self.base()` or `self.base_mut()
+Find and choose the _Player_ node type, which will be a child of Sprite2D in the Change Type dialog that appears.
+
+Now, save your changes, and run the scene.
+The sprite should rotate at a constant speed.
```
Could be on a single line; Markdown will combine it anyway.

Bromeon reviewed (commented):
Thanks a lot, these are great improvements! Added some comments, mostly minor formal things 😊 

---

> >  When exporting your project, you need to use paths inside `res://`.
> > Outside paths like .. are not supported.
>
> This is especially confusing since the tutorial recommends using paths prefixed with `res://../rust/target/`, which is clearly _outside_ the godot resource root folder.
>
> Would this be a better rewording?
>
> > When exporting your project, you need to use `res://` paths. You cannot use filesystem paths.

There is no contradiction because "outside paths" (`res://../something`) can be used during development, and this part talks about _exporting the project_.

I don't think your suggestion is clearer, because `res://../something` _is_ a `res://` path but can still not be used for exporting. The distinction of inside/outside is important, not filesystem or `res://`.

That said, I agree this could be clarified, but maybe it's enough to highlight the difference between development and exporting a bit better.

jpaugh committed:
Update src/intro/hello-world.md

Co-authored-by: Jan Haller <bromeon@gmail.com>

jpaugh committed:
Make changes from code review

- Shorten warning box
- Replace "at `project_dir/some/path`" syntax with English
- Wrap Player and Sprite2D class names in backtics, not italics
- Line wrapping

Bromeon reviewed (approved):
Thanks a lot for your contribution! 🙂
